Solita.LocalizationEditor.UI/Business/BlobLocalizationProviderInitiator.cs

```diff
@@ -0,0 +1,232 @@
+        }
+
+        private BlobLocalizationService GetTranslationProvider()
+        {
+            if (this.LocalizationService == null)
+            {
+                return null;
+            }
+
+            // Gets any provider that has the same name as the one initialized.
+            LocalizationProvider localizationProvider =
+                this.LocalizationService.Providers.FirstOrDefault(
+                    p => p.Name.Equals(ProviderName, StringComparison.Ordinal));
+
+            return localizationProvider as BlobLocalizationService;
+        }
+
+        /// <summary>
+        ///     Loads the provider.
+        /// </summary>
+        /// <returns>
+        ///     [true] if the provider has been loaded.
+        /// </returns>
+        private bool LoadProvider()
+        {
+            if (this.LocalizationService == null)
+            {
+                return false;
+            }
+
+            // This config value could tell the provider where to find the translations, 
+            // set to 0 though, will be looked up after initialization in the provider itself.
+            NameValueCollection configValues = new NameValueCollection { { "containerid", "0" } };
+            var translationProviderProvider = new BlobLocalizationService();
+            translationProviderProvider.Initialize(ProviderName, configValues);
+            try
+            {
+                this.LocalizationService.Providers.Add(translationProviderProvider);
+            }
+            catch (NotSupportedException notSupportedException)
+            {
+                Log.Error("[LocalizationResult] Error add provider to the LocalizationResult Service.", notSupportedException);
+                return false;
+            }
+            LoadTranslations(translationProviderProvider);
+
+            return true;
+        }
+
+        /// <summary>
+        ///     Uns the load provider.
+        /// </summary>
+        /// <returns>
+        ///     [false] if the provider has been unloaded, as it's not initialized anymore.
+        /// </returns>
+        private bool UnLoadProvider(LocalizationProvider localizationProvider)
+        {
+            if (this.LocalizationService == null)
+            {
+                return false;
+            }
+
+            if (localizationProvider != null)
+            {
+                // If found, remove it.
+                this.LocalizationService.Providers.Remove(localizationProvider);
+            }
+
+            return false;
+        }
+
+        public void ReInitProvider()
+        {
+            var service = ServiceLocator.Current.GetInstance<LocalizationService>() as ProviderBasedLocalizationService;
+            if (service == null)
+            {
+                Log.Error("LocalizationService unavailable");
+                return;
+            }
+            var translationProvider = this.GetTranslationProvider();
```

Comment:
"this" keyword is used inconsistently. It is redundant in all of the cases in this file and could be removed.